qCC/ccWaveformDialog.cpp

```diff
@@ -397,15 +399,14 @@ ccWaveDialog::ccWaveDialog(	ccPointCloud* cloud,
 	connect(m_gui->saveWaveToolButton,		&QToolButton::clicked,	this,	&ccWaveDialog::onExportWaveAsCSV);
 	connect(this, &QDialog::finished, [&]() { m_gui->pointPickingToolButton->setChecked(false); }); //auto disable picking mode when the dialog is closed
 
-
 	//force update
 	onPointIndexChanged(0);
 }
 
 ccWaveDialog::~ccWaveDialog()
 {
-
 	delete m_gui;
+	m_display->redraw();
```

Comment:
test m_display

Reply:
Where do you remove the label (if any)?

Reply:
m_label is an std::shared_ptr<cc2DLabel>, the label itself is removed when m_label is deleted, i.e. when the ccWaveformDialog is deleted